### ISSUE TITLE
BeforeLeaveEvent postpone and proceed method's work is unexpected

### ISSUE THREAD

**Issue by alexberazouski** [vaadin/flow#3613]
`postpone` - prevents view change, but Url is changed.
`postpone + proceed` - doesn't navigate to destination view after `proceed`

https://github.com/alexberazouski/skeleton-starter-flow-spring/commit/43e964d40d88e20861bdc41d7ecf16c270de535f

Navigate to: help/postpone; click "To main view". Navigation canceled, but url changed
Navigate to help/postcontinue; click "To main view". Navigation canceled, url changed, but I expect that it should proceed

Also I would expect the postpone call changes back the Url and the browser history returns to the previous state

**Comment by denis-anisimov** [comment]
These are two different issues as I understand:
- `BeforeLeaveEvent::postpone` postpones the navigation but the URL is changed immediately
- `BeforeLeaveEvent::postpone`  with following `BeforeLeaveEvent::proceed` doesn't navigate at all.

If I get it right (that there are two issues) then they are completely different and should not be combined together at all.
Please keep it in mind generally: create a ticket with only one issue. Don't combine them into one even though they look similar and about the same functionality.

I will create the ticket about the first issue. 
Here I'm going to work on the second issue.

There is design issue with `proceed` : it is not supposed to be used immediately. It's supposed that there will be another roundtrip between the client-side and the server-side and only after that `proceed` will be called.
But it's of course a bug in the design: it should not assume anything about a proper usecase. It should work if `proceed` is immediately called. And if it's called afterwards. 

Technically neither first nor the second are not serious issues: the first one won't be fixed now most likely since it makes just a small visual effect which is not critical and the second is just improper usage of API (you don't need to use it in this way at all since it has no any sense).

**Comment by denis-anisimov** [comment]
#3619 is create for the first issue.
